lib/auth.ts

```diff
@@ -381,10 +382,23 @@ export const authOptions: AuthOptions = {
             console.log('Could not find an existing user for the email', email);
             return null;
           }
+          let businessType = undefined;
+
+          if (
```

Comment:
This works, but would be good to do with a `switch` statement, and to leverage a union type that has the possible values of `businessType`, to ensure we are handling all cases. Like so:

```
switch (credentials?.businessType) {
  case 'company':
  case 'individual':
   return credentials?.businessType;
  default:
    assertUnreachable(credentials?.businessType);
    return 'individual'; // We default to individual, but rely on TS to catch missing cases
}
```

Reply:
Added!